src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/costmodel/CFRNumberOfVarsShippedInResponses.java

```diff
@@ -59,24 +56,19 @@ else if ( lop instanceof LogicalOpBGPAdd ) {
 			}
 		}
 		else if ( lop instanceof LogicalOpRequest ) {
-			final DataRetrievalRequest req = ((LogicalOpRequest<?, ?>) lop).getRequest();
+			final CompletableFuture<Integer> futureIntResSize = initiateCardinalityEstimation(plan);
```

Comment:
In this case, `futureIntResSize` should be created not here but only within the if-branch for SPARQL endpoints. The reason for this is that `futureIntResSize` is not needed for TPF/brTPF servers. We shouldn't start a cardinality estimation process if we don't actually need the cardinality estimate ;-)